app/src/main/java/com/teamwss/websoso/ui/feed/FeedFragment.kt

```diff
@@ -1,7 +1,152 @@
+        }
+
+        override fun onThumbUpButtonClick(view: View, id: Int) {
+            feedViewModel.updateLikeCount(view.isSelected, id)
+        }
+
+        override fun onCommentButtonClick() {
+            navigateToFeedDetail(id)
+        }
+    }
+
+    @SuppressLint("InflateParams")
+    private fun showMenu(view: View, feedId: Int, userId: Int) {
+        PopupWindow(
+            popupBinding.root,
+            WindowManager.LayoutParams.WRAP_CONTENT,
+            WindowManager.LayoutParams.WRAP_CONTENT,
+            true
+        ).apply {
+            popupBinding.feedId = feedId
+            popupBinding.userId = userId
+            showAsDropDown(view)
+        }
+    }
+
+    private fun navigateToFeedDetail(id: Int) {
+        startActivity(
+            FeedDetailActivity.from(
+                id, requireContext()
+            )
+        )
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setupAdapter()
+        setupPopupBinding()
+        setupCategory()
+        observeUiState()
+    }
+
+    private fun setupAdapter() {
+        binding.rvFeed.adapter = feedAdapter
+        binding.rvFeed.setHasFixedSize(true)
+    }
+
+    private fun setupPopupBinding() {
+        popupBinding.lifecycleOwner = viewLifecycleOwner
+        popupBinding.viewModel = feedViewModel
+    }
+
+    private fun setupCategory() {
+        when (feedViewModel.gender) {
+            "MALE" -> getString(R.string.feed_category_male).toWrappedCategories()
+            "FEMALE" -> getString(R.string.feed_category_female).toWrappedCategories()
+            else -> throw IllegalStateException()
+        }.apply {
+            setUpChips()
+            binding.wcgFeed.getChildAt(0).performClick()
+        }
+    }
+
+    private fun List<Category>.setUpChips() {
+        forEach { category ->
+            WebsosoChip(requireContext()).apply {
+                setWebsosoChipText(category.title)
+                setWebsosoChipTextAppearance(R.style.title3)
+                setWebsosoChipTextColor(R.color.bg_feed_chip_text_selector)
+                setWebsosoChipStrokeColor(R.color.bg_feed_chip_stroke_selector)
+                setWebsosoChipBackgroundColor(R.color.bg_feed_chip_background_selector)
+                setWebsosoChipPaddingVertical(20f)
+                setWebsosoChipPaddingHorizontal(12f)
+                setWebsosoChipRadius(30f)
+                setOnWebsosoChipClick { feedViewModel.fetchFeedsByCategory(category) }
+            }.also { websosoChip -> binding.wcgFeed.addChip(websosoChip) }
+        }
+    }
+
+    private fun observeUiState() {
+        feedViewModel.uiState.observe(viewLifecycleOwner) { uiState ->
+            if (uiState.loading) loading()
+            if (uiState.error) throw IllegalStateException() // 에러 헨들링 필요
+            if (!uiState.loading) feedAdapter.submitList(uiState.feeds)
+        }
+    }
```

Comment:
```suggestion
private fun observeUiState() {
    feedViewModel.uiState.observe(viewLifecycleOwner) { uiState ->
        when {
            uiState.loading -> loading()
            uiState.error -> throw IllegalStateException("Error handling needed") 
            !uiState.loading -> feedAdapter.submitList(uiState.feeds)
        }
    }
}
```
이렇게 `when`으로 처리하는 것은 별로일까요?

Reply:
ㅋㅋ이거 리뷰달거 같았는데
처음에 아래처럼했다가 바꿨씀다
if문으로 구성해도 깔끔하고 뎁스도 줄길래 ㅎ
추후 다시 손좀보겠습니다